src/parameters/update_cost_parameters.jl

```diff
@@ -114,30 +203,45 @@ function _update_pwl_cost_expression(
     return gen_cost
 end
 
+# For multi-start variables, we need to get a subset of the parameter
+_index_into_param(cost_data, ::T) where {T <: Union{StartVariable, MultiStartVariable}} =
+    start_up_cost(cost_data, T())
+_index_into_param(cost_data, ::VariableType) = cost_data
+
+# General case
+# (TODO this seemed rather decrepit before I got here and made big changes, make sure I didn't break anything)
 function update_variable_cost!(
     container::OptimizationContainer,
-    parameter_array::JuMPFloatArray,
-    parameter_multiplier::JuMPFloatArray,
-    attributes::CostFunctionAttributes{Float64},
-    component::T,
+    parameter_array::DenseAxisArray{T},
+    parameter_multiplier::JuMPFloatArray,  # TODO is the multiplier always `JuMPFloatArray`?
+    attributes::CostFunctionAttributes{T},
+    component::U,
     time_period::Int,
-) where {T <: PSY.Component}
-    resolution = get_resolution(container)
-    dt = Dates.value(resolution) / MILLISECONDS_IN_HOUR
-    base_power = get_base_power(container)
+) where {T, U <: PSY.Component}
     component_name = PSY.get_name(component)
-    cost_data = parameter_array[component_name, time_period]  # TODO is this a new-style cost?
-    if iszero(cost_data)
-        return
-    end
+    cost_data = parameter_array[component_name, time_period]
+    # TODO is this really correct? If it's zero now we can just leave it, it won't keep the value from last round?
```

Comment:
This function is getting called for each time period, if the cost term is zero, then why bother adding it to the variant terms or the cost expression